src/lightning_app/cli/lightning_cli.py

```diff
@@ -232,7 +232,12 @@ def _run_app(
     secret: tuple,
     run_app_comment_commands: bool,
 ) -> None:
-    file = _prepare_file(file)
+
+    if not os.path.exists(file):
```

Comment:
if user makes a mistake in the file name, `_validate_name` from the `gallery_apps_and_components` will throw weird errors. Can do valid name check before this `exists` check and raise nice error message

Reply:
Resolved this by failing later on.

Reply:
It wouldn't reach "later on" if `_valid_name` function fails. Like try `directory/wrongfile.py`?